spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerArgumentAdapter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binding;
+
+import org.springframework.core.MethodParameter;
+
+/**
+ * Strategy for adapting a method argument type annotated with
+ * {@link org.springframework.cloud.stream.annotation.Input} or
+ * {@link org.springframework.cloud.stream.annotation.Output} from a bound element
+ * (e.g. {@link org.springframework.messaging.MessageChannel} supported by an
+ * existing binder.
+ *
+ * This is a framework extension and is not primarily intended for use by end-users.
+ * @author Marius Bogoevici
+ */
+public interface StreamListenerArgumentAdapter<A, B> {
```

Comment:
based on the comments and usage, I'm actually wondering if this should be renamed to `StreamListenerParameterAdapter` ("parameter" instead of "argument")